packages/babel-parser/src/parser/expression.js

```diff
@@ -2479,77 +2479,47 @@ export default class ExpressionParser extends LValParser {
     startPos: number,
     startLoc: Position,
   ): N.PipelineBody {
-    const pipelineStyle = this.checkSmartPipelineBodyStyle(childExpression);
-
-    this.checkSmartPipelineBodyEarlyErrors(
-      childExpression,
-      pipelineStyle,
-      startPos,
-    );
+    this.checkSmartPipelineBodyEarlyErrors(childExpression, startPos);
 
     return this.parseSmartPipelineBodyInStyle(
       childExpression,
-      pipelineStyle,
       startPos,
       startLoc,
     );
   }
 
   checkSmartPipelineBodyEarlyErrors(
     childExpression: N.Expression,
-    pipelineStyle: N.PipelineStyle,
     startPos: number,
   ): void {
     if (this.match(tt.arrow)) {
       // If the following token is invalidly `=>`, then throw a human-friendly error
       // instead of something like 'Unexpected token, expected ";"'.
       throw this.raise(this.state.start, Errors.PipelineBodyNoArrow);
-    } else if (
-      pipelineStyle === "PipelineTopicExpression" &&
-      childExpression.type === "SequenceExpression"
```

Comment:
`childExpression.type === "SequenceExpression"` implies the fact that `pipelineStyle === "PipelineTopicExpression"`.